modules/enhancer.js

```diff
@@ -2,40 +2,39 @@
 
 var resolveStyles = require('./resolve-styles.js');
 var wrapUtils = require('./wrap-utils.js');
+var assign = require('lodash/object/assign');
 
 var enhanceWithRadium = function (ComposedComponent) {
-  var RadiumEnhancer = function () {
-    ComposedComponent.prototype.constructor.call(this);
 
-    if (!this.state) {
-      this.state = {};
-    }
+  class RadiumEnhancer extends ComposedComponent {
+
+    constructor (props) {
+      super(props);
 
-    var radiumInitialState = wrapUtils.getInitialState();
-    Object.keys(radiumInitialState).forEach(function (key) {
-      this.state[key] = radiumInitialState[key];
-    }, this);
-  };
+      var radiumInitialState = wrapUtils.getInitialState();
+      this.state = assign(this.state || {}, radiumInitialState);
+    }
 
-  RadiumEnhancer.prototype = new ComposedComponent();
-  RadiumEnhancer.prototype.constructor = RadiumEnhancer;
+    render () {
+      var renderedElement = super.render();
+      return resolveStyles(this, renderedElement);
+    }
 
-  RadiumEnhancer.prototype.render = function () {
-    var renderedElement = ComposedComponent.prototype.render.call(this);
-    return resolveStyles(this, renderedElement);
-  };
+    componentWillUnmount () {
+      if (super.componentWillUnmount) {
+        super.componentWillUnmount();
+      }
 
-  RadiumEnhancer.prototype.componentWillUnmount = function () {
-    if (ComposedComponent.prototype.componentWillUnmount) {
-      ComposedComponent.prototype.componentWillUnmount.call(this);
+      wrapUtils.componentWillUnmount(this);
     }
+  }
 
-    wrapUtils.componentWillUnmount(this);
-  };
+  const displayName =
+    ComposedComponent.displayName ||
+    ComposedComponent.name ||
+    'Component';
 
-  RadiumEnhancer.defaultProps = ComposedComponent.defaultProps;
-  RadiumEnhancer.propTypes = ComposedComponent.propTypes;
-  RadiumEnhancer.contextTypes = ComposedComponent.contextTypes;
+  RadiumEnhancer.displayName = `Radium(${displayName})`;
```

Comment:
Afaik, the static props should be inherited by default. [ie](http://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&code=class%20A%20%7B%0A%20%20static%20X%20%3D%201%0A%7D%0A%0Aclass%20B%20extends%20A%20%7B%7D%0A%0Aconsole.log(B.X)%2F%2F1)